src/client/components/dashboard/SettingsModal.tsx

```diff
@@ -0,0 +1,412 @@
+                <Switch
+                  id="isPublished"
+                  isChecked={isActive}
+                  colorScheme="teal"
+                  isDisabled={isDisabled}
+                  onChange={onSwitchChange}
+                />
+              </HStack>
+            </FormControl>
+            <Divider />
+
+            <FormControl>
+              <FormLabel htmlFor="share">
+                Share link
+                <Text color="grey" fontSize="xs">
+                  Create an official short link and QR code with{' '}
+                  <Link color="blue.700" href="https://go.gov.sg/#/">
+                    Go.gov.sg
+                  </Link>{' '}
+                  and share it over the Internet
+                </Text>
+              </FormLabel>
+              <HStack spacing={1} align="stretch">
+                <EmbedField name="iframe" value={linkToChecker} />
+                <Spacer />
+                <DefaultTooltip
+                  label={
+                    isActive
+                      ? `Open live checker`
+                      : `Publish and activate your checker first`
+                  }
+                >
+                  <Link href={isActive ? linkToChecker : undefined} isExternal>
+                    <IconButton
+                      colorScheme="primary"
+                      aria-label="Open published form link"
+                      icon={<BiLinkExternal />}
+                      type="submit"
+                      rounded="md"
+                      isDisabled={!isActive}
+                    />
+                  </Link>
+                </DefaultTooltip>
+              </HStack>
+            </FormControl>
+            <Divider />
+
+            <FormControl>
+              <FormLabel htmlFor="embed">
+                Embed HTML
+                <Text color="grey" fontSize="xs">
+                  Embed this checker on your site
+                </Text>
+              </FormLabel>
+              <EmbedField
+                name="iframe"
+                value={`<iframe src="${linkToChecker}" style="width:100%;height:500px"></iframe>`}
+              />
+            </FormControl>
+            <Divider />
+          </VStack>
+        </ModalBody>
+
+        <ModalHeader>Manage Collaborators</ModalHeader>
+        <ModalBody>
+          <VStack spacing={4}>
+            <FormControl>
+              <FormLabel htmlFor="collaborators">
+                Add collaborators or transfer checker ownership
+                <Text color="grey" fontSize="xs">
+                  Only users that have logged in to CheckFirst before with a
+                  government email may be added as collaborators
+                </Text>
+              </FormLabel>
+              <HStack spacing={2}>
+                <Input
+                  placeholder="me@example.gov.sg"
+                  value={inputEmail}
+                  onChange={handleInputChange}
+                />
+                <Button
```

Comment:
Disabled the button when the value for `inputEmail` is empty. You can also add the `isLoading` prop when the request is being processed to give immediate feedback to the user after the button is clicked.

![image](https://user-images.githubusercontent.com/3666479/126948108-d91503cd-acc5-4cf6-ad97-caef8b4c1ffd.png)

Reply:
Add collaborator button is now disabled when email input field is empty. Also sets isLoading when button is clicked.